scripts/msct_register.py

```diff
@@ -153,34 +218,55 @@ def register2d_centermassrot(fname_src, fname_dest, fname_warp='warp_forward.nii
-            # compute (src,dest) angle for first eigenvector
-            if rot == 1:
-                eigenv_src = pca_src[iz].components_.T[0][0], pca_src[iz].components_.T[1][0]  # pca_src.components_.T[0]
-                eigenv_dest = pca_dest[iz].components_.T[0][0], pca_dest[iz].components_.T[1][0]  # pca_dest.components_.T[0]
-                angle_src_dest[iz] = angle_between(eigenv_src, eigenv_dest)
-                # check if ratio between the two eigenvectors is high enough to prevent poor robustness
-                if pca_src[iz].explained_variance_ratio_[0] / pca_src[iz].explained_variance_ratio_[1] < pca_eigenratio_th:
-                    angle_src_dest[iz] = 0
-                if pca_dest[iz].explained_variance_ratio_[0] / pca_dest[iz].explained_variance_ratio_[1] < pca_eigenratio_th:
-                    angle_src_dest[iz] = 0
-            # append to list of z_nonzero
-            z_nonzero.append(iz)
-        # if one of the slice is empty, ignore it
-        except ValueError:
-            sct.printv('WARNING: Slice #' + str(iz) + ' is empty. It will be ignored.', verbose, 'warning')
+
+    if rot <= 1:
+        # Loop across slices
+        for iz in range(0, nz):
+            try:
+                # compute PCA and get center or mass
+                coord_src[iz], pca_src[iz], centermass_src[iz, :] = compute_pca(data_src[:, :, iz])
+                coord_dest[iz], pca_dest[iz], centermass_dest[iz, :] = compute_pca(data_dest[:, :, iz])
+                # compute (src,dest) angle for first eigenvector
+                if rot == 1:
+                    eigenv_src = pca_src[iz].components_.T[0][0], pca_src[iz].components_.T[1][0]  # pca_src.components_.T[0]
+                    eigenv_dest = pca_dest[iz].components_.T[0][0], pca_dest[iz].components_.T[1][0]  # pca_dest.components_.T[0]
+                    angle_src_dest[iz] = angle_between(eigenv_src, eigenv_dest)
+                    # check if ratio between the two eigenvectors is high enough to prevent poor robustness
+                    if pca_src[iz].explained_variance_ratio_[0] / pca_src[iz].explained_variance_ratio_[1] < pca_eigenratio_th:
+                        angle_src_dest[iz] = 0
+                    if pca_dest[iz].explained_variance_ratio_[0] / pca_dest[iz].explained_variance_ratio_[1] < pca_eigenratio_th:
+                        angle_src_dest[iz] = 0
+                # append to list of z_nonzero
+                z_nonzero.append(iz)
+            # if one of the slice is empty, ignore it
+            except ValueError:
+                sct.printv('WARNING: Slice #' + str(iz) + ' is empty. It will be ignored.', verbose, 'warning')
+
+    else:  # im and seg case
+        for iz in range(0, nz):
+            try:
+                _, _, centermass_src[iz, :] = compute_pca(data_src_seg[:, :, iz])
+                _, _, centermass_dest[iz, :] = compute_pca(data_dest_seg[:, :, iz])
+
+                # TODO: Here will be put the new method to find the angle
```

Comment:
i didn't realize your method was not implemented in this PR. It's ok, but just be clear about it. Once this PR is merged, the method HOG will be available from the usage, but not implemented. So in the meantime, you should do something like `raise NotImplementedError`